### ISSUE TITLE
Fix for Generic.Functions.OpeningFunctionBraceKernighanRitchie.BraceOnNewLine leaves blank line following the brace

### ISSUE THREAD

**Issue by jasonmccreary** [squizlabs/PHP_CodeSniffer#1085]
The `Squiz.Functions.MultiLineFunctionDeclaration.BraceOnNewLine` fix should condense extra line breaks following the opening brace.

### Ran

    vendor/bin/phpcbf --standard=PSR2 given.php

### given.php

```php
$numbers = [1, 2, 3];
array_walk($numbers, function()
{
    return 5;
});
```

### Expected

```php
$numbers = [1, 2, 3];
array_walk($numbers, function () {
    return 5;
});
```

### Actual

```php
$numbers = [1, 2, 3];
array_walk($numbers, function () {

    return 5;
});
```

Realize this is minor. But if it's inline with others expectations, glad to submit a fix.

Great tool. Thank you.

**Comment by gsherwood** [comment]
Technically, it shouldn't remove all the extra newlines it finds after the brace because that isn't part of what the sniff is checking. In PSR2, for example, you can have as many blank lines as you'd like there.

But what it should not be doing is adding more blank lines than were already there, and that's exactly what is happening in your case. The sniff is making the assumption that other parts of your standard will fix the blank line problem, but that's not the case in PSR2 (at least).

**Comment by gsherwood** [comment]
I recorded this as an enhancement because the current behaviour was expected and coded into the unit test, so this was more the sniff being lazy than not working.

Either way, it is fixed now. It also had to remove any indent on that first line in the block or else the first line would get a double indent.

**Comment by jasonmccreary** [comment]
Make sense.

I actually wrote a patch myself for the fun. But yours looks better ;)

**Comment by gsherwood** [comment]
> I actually wrote a patch myself for the fun.

I don't normally beat people to the patch, but had an hour free this morning to work on PHPCS :)

Maybe I'll see a PR from you next time ;)

**Comment by jasonmccreary** [comment]
I was reluctant as I didn't feel it was a strictly a *bug*. However, I use this tool a lot. Glad to contribute anytime.
